admin/includes/classes/stats_sales_report_graph.php

```diff
@@ -1,38 +1,46 @@
 <?php
 /**
  * @package admin
- * @copyright Copyright 2003-2016 Zen Cart Development Team
+ * @copyright Copyright 2003-2018 Zen Cart Development Team
  * @author inspired from sales_report_graphs.php,v 0.01 2002/11/27 19:02:22 cwi Exp  Released under the GNU General Public License $
  * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
- * @version $Id:  New in v1.6.0 $
+ * @version $Id: lat9 Tue Sep 25 09:35:40 2018 -0400 New in v1.5.6 $
  */
 
 class statsSalesReportGraph {
 
-  var $mode, $globalStartDate, $startDate, $endDate, $info, $previous, $next, $startDates, $endDates, $size;
+  protected $mode = 4;
+  protected $globalStartDate, $startDate, $endDate, $startDates, $endDates;
+  public $info = array();
+  public $previous, $next, $filter = '';
+  public $size = 0;
 
-  function __construct($mode, $startDate = "", $endDate = "", $filter = "") {
-    // startDate and endDate have to be a unix timestamp. Use mktime !
-    // if set then both have to be valid startDate and endDate
+    /**
+     * statsSalesReportGraph constructor.
+     *
+     * startDate and endDate have to be a unix timestamp. Use mktime !
+     * if set then both have to be valid startDate and endDate
+     *
+     * @param integer $mode number indicating report format
+     * @param string $startDate unix timestamp
+     * @param string $endDate unix timestamp
+     * @param string $filter filter string
+     */
+  public function __construct($mode, $startDate = '', $endDate = '', $filter = '') {
     global $db;
     $this->mode = $mode;
-    $this->previous = "";
-    $this->next = "";
-    $this->filter = "";
-    $this->info = array(array());
     // get date of first sale
     $first = $db->Execute("select UNIX_TIMESTAMP(min(date_purchased)) as first FROM " . TABLE_ORDERS);
-    $this->globalStartDate = mktime(0, 0, 0, date("m", $first->first), date("d", $first->first), date("Y", $first->first));
+    $this->globalStartDate = mktime(0, 0, 0, date("m", $first->fields['first']), date("d", $first->fields['first']), date("Y", $first->fields['first']));
```

Comment:
Removed.

Reply:
Sorry, I think that went the wrong direction. Did you test that it works this way?

Since `$first` is the query result, we can access its properties, either as `$first->fields['property']` or inside a foreach loop as `$first['property']`. 
It looks like there are mixed implementations here, so I'm surprised if this works.

Reply:
I'm okay if the ->fields approach is used. I just spotted a mixed use of things, and thought it should be made consistent.